lib/plugins/aws/deploy/lib/createStack.js

```diff
@@ -93,15 +93,14 @@ module.exports = {
 
     // check if the user has added some "custom provider resources"
     // and merge them into the CloudFormation template if there are any
-    if (this.serverless.service.resources && this.serverless.service.resources.Resources) {
-      _.forEach(this.serverless.service.resources.Resources, (value, key) => {
-        const newResourceObject = {
-          [key]: value,
-        };
 
-        _.merge(coreCloudFormationTemplate.Resources, newResourceObject);
-      });
-    }
+    _.forEach(this.serverless.service.resources, (resourceValue, resourceKey) => {
+      if (typeof resourceValue === 'string') {
+        coreCloudFormationTemplate[resourceKey] = resourceValue;
+      } else if (typeof resourceValue === 'object') {
```

Comment:
This if-else check is used so that users can overwrite properties such as the `Description` or the `AWSTemplateFormatVersion` (which are strings). Other properties such as `Resources` or `Outputs` are objects.